src/dswx_sar/detect_inundated_vegetation.py

```diff
@@ -36,16 +51,32 @@ def run(cfg):
     inundated_vege_ratio_threshold = \
         inundated_vege_cfg.dual_pol_ratio_threshold
     inundated_vege_cross_pol_min = inundated_vege_cfg.cross_pol_min
-    inundated_vege_target = inundated_vege_cfg.target_land_cover
+
+    target_file_type = inundated_vege_cfg.target_area_file_type
+    target_worldcover_class = inundated_vege_cfg.target_worldcover_class
+    target_glad_class = inundated_vege_cfg.target_glad_class
 
     line_per_block = inundated_vege_cfg.line_per_block
-    filter_options = processing_cfg.filter
-    filter_method = processing_cfg.filter.method
+    filter_options = inundated_vege_cfg.filter
+    filter_method = inundated_vege_cfg.filter.method
+
+    interp_glad_path_str = os.path.join(outputdir, 'interpolated_glad.tif')
+    interp_worldcover_path_str = os.path.join(outputdir,
+                                              'interpolated_landcover.tif')
+
+    if (target_file_type == 'auto') and os.path.exists(interp_glad_path_str):
+        target_file_type = 'GLAD'
+    else:
+        target_file_type = 'WorldCover'
 
     # Currently, inundated vegetation for C-band is available for
     # Herbanceous wetland area
-    landcover_path_str = os.path.join(outputdir, 'interpolated_landcover.tif')
-    mask_obj = FillMaskLandCover(landcover_path_str)
+    if target_file_type == 'WorldCover':
+        landcover_path_str = interp_worldcover_path_str
+    else:
+        landcover_path_str = interp_glad_path_str
```

Comment:
I think it would be better not to move the paths. `interp_glad_path_str` and `interp_worldcover_path_str` are being used two time in line 87, 98 and line 96, 99, respectively.